AdobeStockAsset/Model/Request/Builder.php

```diff
@@ -0,0 +1,127 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+namespace Magento\AdobeStockAsset\Model\Request;
+
+use Magento\AdobeStockAssetApi\Api\RequestBuilderInterface;
+use Magento\AdobeStockAsset\Model\Request\Builder\ConfigInterface;
+use Magento\Framework\Phrase;
+use Magento\Framework\Search\Request\NonExistingRequestNameException;
+use Magento\AdobeStockAssetApi\Api\Data\RequestInterfaceFactory;
+
+/**
+ * Class Builder
+ * @package Magento\AdobeStockAsset\Model\Request
+ */
+class Builder implements RequestBuilderInterface
+{
+    /**
+     * @var ConfigInterface
+     */
+    private $config;
+
+    /**
+     * @var array
+     */
+    private $data;
+
+    /**
+     * @var Builder\Binder
+     */
+    private $binder;
+
+    /**
+     * @var RequestInterfaceFactory
+     */
+    private $requestFactory;
+
+    /**
+     * Builder constructor.
+     * @param ConfigInterface $config
+     * @param Builder\Binder $binder
+     * @param RequestInterfaceFactory $requestFactory
+     */
+    public function __construct(
+        ConfigInterface $config,
+        Builder\Binder $binder,
+        RequestInterfaceFactory $requestFactory
+    ) {
+        $this->config = $config;
+        $this->binder = $binder;
+        $this->requestFactory = $requestFactory;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function setName(string $name) : void
+    {
+        $this->data['requestName'] = $name;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function setSize(int $size): void
+    {
+        $this->data['size'] = $size;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function setOffset(int $offset): void
+    {
+        $this->data['offset'] = $offset;
```

Comment:
Considering that these keys are used in builder and binder classes and will be used in request class, we should declare corresponding constants in the RequestInterface

Reply:
Added todo to the builder class